### PR TITLE
Remove jQuery from browser extension components

### PR DESCRIPTION
**Pull request by kashifatcha**
## What
Going through the codebase and finding where jQuery has been used in the browser extension's components and replacing it with vanilla JavaScript. The tests also included jQuery and needed to be migrated over as well. 

## Why
As we are migrating the manifest version of the browser extension from 2 to 3, one of the requirements, it would have been difficult to do this whilst the code was still written in jQuery. Also, as jQuery had been removed from most of Gov.uk, this was a good opportunity to remove it from the extension as well. This was the first time that we had worked on the repo, and it allowed us to better understand the codebase and how it works.

https://trello.com/c/3KgjI5Cr/2178-remove-jquery

### CODE REVIEW

**Review by jon-kirwan — COMMENTED** [review]
Great work on this!

Added comments below. Also, just a suggestion, which is entirely separate to the task, but I wonder if we should add some JavaScript linting to enforce stylistic rules like single quotes, removing empty line spaces, semi-colon use etc. Many GOV.UK applications include a dependency "standardx" which can be added to `package.json` file with: `yarn add standardx`.

And may want to enforce rules like so:
https://github.com/alphagov/govuk_publishing_components/blob/main/package.json#L21-L33

File: src/components/highlight-component/highlight-component.js [hunk 1]
```diff
@@ -49,9 +64,9 @@ function HighlightComponent() {
 
 HighlightComponent.prototype.toggleComponents = function () {
   this.state = !this.state;
-
+  // TODO: Reformat this loop to make it more readable.
```
Comment (KludgeKML) [inline src/components/highlight-component/highlight-component.js]: This loop feels okay in terms of readability? You could maybe remove the comment.

**Comment by KludgeKML** [comment]
Looks okay to me, if you address Jon's comments.

**Comment by KludgeKML** [comment]
@jon-kirwan is there a coverage tool that works with Jasmine? (ie can @kashifatcha see what code coverage this version has compared to the old version)

**Comment by jon-kirwan** [comment]
> @jon-kirwan is there a coverage tool that works with Jasmine? (ie can @kashifatcha see what code coverage this version has compared to the old version)

Interesting question. I don't know much but a quick search around seems to suggest that [Istanbul](https://github.com/istanbuljs/nyc) is one of the more popular tools for measuring JavaScript code coverage and can be integrated with Jasmine.

There's an open issue on the Design System repo which mentions it too: [expand code coverage collection ](https://github.com/alphagov/govuk-frontend/issues/3101).

Might be a good one to add to the backlog to explore further.

File: src/components/highlight-component/highlight-component.js (hunk 1 above)
Comment (kashifatcha) [reply src/components/highlight-component/highlight-component.js]: Ahhh yes, I had quite a number of these TODO comments and missed this one out. I'll remove it :)

File: spec/support/jasmine-browser.json [hunk 2]
```diff
@@ -1,9 +1,35 @@
...
-  "specFiles": ["**/show_meta_tags_spec.js", "**/highlight_component_spec.js", "**/extract_path_spec.js", "**/external_links_spec.js", "**/environment_spec.js", "**/design_mode_component_spec.js", "**/content_links_spec.js", "**/ab_tests_spec.js", "**/ab_bucket_store_spec.js"],
-  "helpers": ["helpers/jquery.min.js", "helpers/jasmine-jquery.js", "helpers/helpers.js", "helpers/polyfills/String.startsWith.js"],
+  "specFiles": [
+    "**/show_meta_tags.spec.js",
+    "**/highlight_component.spec.js",
+    "**/extract_path_spec.js",
+    "**/external_links_spec.js",
+    "**/environment_spec.js",
+    "**/design_mode_component.spec.js",
+    "**/content_links_spec.js",
```
[the remark this answers is not in the document]
Comment (jon-kirwan) [reply spec/support/jasmine-browser.json:23]: Thanks Kash, good spot. I checked some other repos too. In `collections` the specs have a `_spec` extension https://github.com/alphagov/collections/tree/main/spec/javascripts/modules but in `government-frontend` the naming convention is `.spec.js` https://github.com/alphagov/government-frontend/tree/main/spec/javascripts/modules. Jasmine doesn't seem to have any guidance on the best practice. But yeah probably the best thing to do is keep the naming convention the same (either naming style seems fine to me).

**Comment by kashifatcha** [comment]
I have made the changes suggested in the review. I will create a separate PR for adding the linter.

**Comment by jon-kirwan** [comment]
> I have made the changes suggested in the review. I will create a separate PR for adding the linter.

Thanks Kash. The changes look great and all work fine for me. I just have two final commit history related suggestions:
- you might be able to squash some commits together e.g. you can probably remove the commit for `Fix staleReferenceElementError in design-mode-component spec file` (that can probably be squashed into one for rewriting the spec files in JavaScript)
- add a separate commit for the change to `manifest.json`, to update the version. Also, in the same commit, run `npm run build ` to create a new package with the changes. More info: https://github.com/alphagov/govuk-browser-extension?tab=readme-ov-file#releasing-the-extension.

If you don't mind taking a look at these changes i'll then hit approve.

**Comment by kashifatcha** [comment]
I have squashed the commits, so this PR should be ready to go. I will submit a separate PR for the linter and for upgrading to manifest V3.

**Review by jon-kirwan — APPROVED** [review]
Well done. All looks good to me!

**Pull request merged**